### PR TITLE
Allow to stack mask_fit in Dataset.stack

### PR DESCRIPTION
**Pull request by QRemy**
Allow to stack mask_fit in Dataset.stack in the same way than mask_safe

### CODE REVIEW

File: gammapy/datasets/map.py [hunk 1]
```diff
@@ -786,6 +786,11 @@ def stack(self, other, nan_to_num=True):
         if self.mask_safe and other.mask_safe:
             self.mask_safe.stack(other.mask_safe)
 
+        if self.mask_fit and other.mask_fit:
+            self.mask_fit.stack(other.mask_fit)
+        elif other.mask_fit:
+            self.mask_fit = other.mask_fit
```
Comment (adonath) [inline gammapy/datasets/map.py:792]:
```suggestion
            self.mask_fit = other.mask_fit.copy()
```

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @QRemy, I have left one inline suggestion. I think we should copy the other mask...

**Review by registerrier — APPROVED** [review]
Thanks @QRemy

**Comment by registerrier** [comment]
@adonath comment has been taken into account. Merging now.

**Pull request merged**